plugins/kubernetes/app/models/kuber_deploy_service.rb

```diff
@@ -46,6 +47,20 @@ def create_services!
     end
   end
 
+  def delete_replication_controllers!
+    kuber_release.release_docs.each do |release_doc|
+      log 'deleting existing ReplicationControllers', role: release_doc.kubernetes_role.name
+      labels = {project: release_group.build.project_name, role: release_doc.kubernetes_role.label_name}
+      client.get_replication_controllers(
+          label_selector: labels.to_kuber_selector,
+          namespace: kuber_release.deploy_group.cluster_deploy_group.namespace).each do |rc|
+        rc.spec.replicas = 0
+        client.update_replication_controller rc
+        client.delete_replication_controller rc.metadata['name'], rc.metadata['namespace']
```

Comment:
I've double checked and we actually need to verify that the number of replicas is 0 before we delete the RC. I'll fix this when we have proper event monitoring in place. BTW, K8S 1.1 should contain graceful pod termination - maybe that will handle the issues we're facing here.